crates/bdk/src/wallet/mod.rs

```diff
@@ -456,6 +459,7 @@ impl<D> Wallet<D> {
         let secp = Secp256k1::new();
         let (chain, chain_changeset) = LocalChain::from_genesis_hash(genesis_hash);
         let mut index = KeychainTxOutIndex::<KeychainKind>::default();
+        index.set_lookahead_for_all(DEFAULT_LOOKAHEAD);
```

Comment:
I created a small test and found you need to move where you set the lookahead to below the `create_signers()` function. But then some other tests began to fail that we'll need to dig into a bit. The changes and test are here: https://github.com/bitcoindevkit/bdk/pull/1230/commits/9d3cb2b3e2907a9fefa5e5389d524daaa752aa99. Feel free to copy/paste them into your PR or cherry pick and squash the commit into yours.

Reply:
Thanks for pointing this out.

The `create_signers` method calls `KeychainTxOutIndex::add_keychain`. `KeychainTxOutIndex::set_lookahead_for_all` method sets the lookaheads for already-added keychains. So if you call `add_keychain` after `set_lookahead_for_all`, the keychain added will always start with a lookahead of 0.

This can be a confusing API. I propose we add an additional parameter to the `add_keychain` method:

```rust
pub fn add_keychain(&mut self, keychain: K, descriptor: Descriptor<DescriptorPublicKey>, lookahead: u32) {
    todo!()
}
```

Reply:
I like this solution, it also makes it more obvious that if you're adding a keychain you should think about the look ahead.

Reply:
Note: it seems the `txout_index` logic is assuming the lookahead is never reduced (for instance it's asserted here https://github.com/bitcoindevkit/bdk/blob/55b680c194b218061254ea898d892c9a5e4cdd3b/crates/chain/src/keychain/txout_index.rs#L393-L396 and here https://github.com/bitcoindevkit/bdk/blob/55b680c194b218061254ea898d892c9a5e4cdd3b/crates/chain/src/keychain/txout_index.rs#L414).

As i introduce a way to set the lookindex in `add_keychain` as suggested above, i'm going to remove `set_lookahead` to prevent this footgun. The alternative obviously would be to drop the asserts and make sure the code doesn't actually assumes this but:
1. This sounds like a lot more work
2. I'm not familiar enough with this code to make this assessment

Does that make sense?

Reply:
Alright so i've implemented @evanlinjin's suggestion and simply modified the first assertion from above to be a `>=` instead of an equality check. I've left out the commit which drops the methods to set the lookahead externally (https://github.com/darosior/bdk/commit/4166ea5e37c859f2f07e33b69c08d183c65ded4c) but i'm still interested in your thoughts about that.

Reply:
It don't think it should be persisted. It's a runtime policy thing and is more dependent on your hardware than it is on your keychain. I'd just document that these methods can only increase it.

Reply:
I will be okay with just asserting that the lookahead cannot decrease.